core/src/main/java/com/graphhopper/routing/util/spatialrules/SpatialRuleLookupArray.java

```diff
@@ -20,20 +20,18 @@
 import com.graphhopper.util.shapes.BBox;
 import com.graphhopper.util.shapes.GHPoint;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.*;
+
+import static com.graphhopper.routing.util.spatialrules.SpatialRule.EMPTY;
 
 /**
- * SpatialRuleLookup implementation using an array as datastructure.
- * Currently limited to 255 rules
- * <p>
+ * SpatialRuleLookup implementation using an array as data structure. Currently limited to 255 ruleContainers
  * The covered area is indexed as tiles, with every tile being "quadratic" having the same degree length on every side.
- * <p>
- * Currently one one rule per Tile is allowed. The later added rule will be added to the tile.
+ * Currently only a limited amount of rule combinations per tile are allowed.
```

Comment:
It is not limited to the number of rules per container but to the number of rule combinations. E.g. A,B is one combination then A,B,D is another and so on. So maybe not that clear description?